content/blogposts/2023/copilot-vs-cody.md

```diff
@@ -0,0 +1,229 @@
+
+![Cody vs Copilot autocomplete](https://storage.googleapis.com/sourcegraph-assets/blog/github-copilot-vs-cody-context-driven-development/copilot-vs-cody-autocomplete.png)
+
+Cody gave us the suggested completion of `$video->avg_frame_rate` from the start. This was exactly the property we were looking for so I could just hit the tab button on my keyboard to accept the completion and move on.
+
+Copilot on the other hand struggled. At first, it suggested a hard coded `30` as the suggestion. So I started typing `$video` to give it a hint that I'm going to be getting this value from the video object. Then Copilot suggested the completion of `->fps` which does not exist on the `$video` object. So if I were to accept this suggestion I would start seeing red squiggly lines telling me something is wrong. Only when I basically completed the line myself did Copilot provide the final part to complete my code.
+
+This really highlights the importance of having the right context. With the right context, a code AI assistant can be a helpful ally, but with bad or the wrong context, it causes more trouble than it's worth. If you'd like to learn more about the lifecycle of a Cody code completion, check out [this in-depth blog post](https://about.sourcegraph.com/blog/the-lifecycle-of-a-code-ai-completion).
+
+Moving forward, when I navigated to the next line, both Cody and Copilot gave me the next autocompletion to convert the `$beginInMilliseconds` value to a timecode format exactly the same.
+
+![Cody and Copilot correct autocomplete suggestion](https://storage.googleapis.com/sourcegraph-assets/blog/github-copilot-vs-cody-context-driven-development/next-autocomplete.png)
+
+Testing this code works and we're able to successfully convert our milliseconds to a timecode format that will allow our API to trim the video at the precise time we declare. For our final autocompletion test, let's see if our code AI assistant can help us with the next steps which will be to call the [AWS Elemental MediaConvert](https://aws.amazon.com/mediaconvert/) API to process the video trimming.
+
+I will start us off by adding a comment `// Call AWS Elemental Media Convert API to trim video` will hit enter and observe what each code AI assistant does.
+
+![Media Convert autocomplete](https://storage.googleapis.com/sourcegraph-assets/blog/github-copilot-vs-cody-context-driven-development/media-convert-autocomplete.png)
+
+Since we make extensive use of the AWS Elemental MediaConvert API, Cody was able to find how we've invoked it in other actions, and gave us helpful completions every step of the way. What really blew me away was the fact that it also found and suggested the required endpoint in our AWS client configuration for this API.
```

Comment:
```suggestion
Since we make extensive use of the AWS Elemental MediaConvert API, Cody was able to find how we've invoked it in other actions and gave us helpful completions every step of the way. What really blew me away was the fact that it also found and suggested the required endpoint in our AWS client configuration for this API.
```